src/test/java/com/ioet/bpm/organizations/BpmOrganizationsApiApplicationTests.java

```diff
@@ -0,0 +1,149 @@
+
+
+    @Test
+    public void createOrganizationResponseSuccess() {
+        Organization organizationCreated = mock(Organization.class);
+        Organization organizationToCreate = mock(Organization.class);
+
+        ResponseEntity<Organization> organizationResponse = organizationController.createOrganization(organizationToCreate);
+
+        verify(organizationRepository).save(organizationToCreate);
+        assertEquals(HttpStatus.CREATED, organizationResponse.getStatusCode());
+
+    }
+
+
+    @Test
+    public void testGetAllOrganizationsUseTheRepository() {
+
+        ResponseEntity<Iterable> organizations = organizationController.getAllOrganizations();
+
+        assertEquals(HttpStatus.OK, organizations.getStatusCode());
+
+        verify(organizationRepository).findAll();
+    }
+
+    @Test
+    public void testGetOrganizationByIdReturnAOrganization() {
+        Organization organizationCreated = mock(Organization.class);
+        Organization organizationToCreate = new Organization();
+        organizationToCreate.setId("id");
+        organizationToCreate.setName("ioet");
+        when(organizationRepository.save(organizationToCreate)).thenReturn(organizationCreated);
+
+        ResponseEntity<Organization> organizationCreatedResponse = organizationController.createOrganization(organizationToCreate);
+
+        assertEquals(organizationCreated, organizationCreatedResponse.getBody());
+        assertEquals(HttpStatus.CREATED, organizationCreatedResponse.getStatusCode());
+        verify(organizationRepository).save(organizationToCreate);
+    }
+
+    @Test
+    public void whenAOrganizationIsFoundThenReturn200() {
+        Organization organizationFound = mock(Organization.class);
+        String organizationFoundId = "id";
+        when(organizationRepository.findById(organizationFoundId)).thenReturn(Optional.of(organizationFound));
+
+        ResponseEntity response = organizationController.getOrganization(organizationFoundId);
+
+
+        verify(organizationRepository).findById(organizationFoundId);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+    }
+
+    @Test
+    public void whenAOrganizationIsNotFoundThenReturn400() {
+        String id = "id";
+        when(organizationRepository.findById(id)).thenReturn(Optional.empty());
+
+        ResponseEntity response = organizationController.getOrganization(id);
+
+
+        verify(organizationRepository).findById(id);
+        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+    }
+
+    @Test
+    public void whenANotFoundOrganizationIsDeletedThenReturn404() {
+        String id = "id";
+        when(organizationRepository.findById(id)).thenReturn(Optional.empty());
+
+        ResponseEntity deleteResponse = organizationController.deleteOrganization(id);
+
+        assertEquals(HttpStatus.NOT_FOUND, deleteResponse.getStatusCode());
+    }
+
+    @Test
+    public void whenAOrganizationIsDeletedAndIsFoundedThenReturnEmpty(){
```

Comment:
Founded does not exist, I think you mean Found

Reply:
I will be more careful. It's done